web_poet/page_inputs/stats.py

```diff
@@ -0,0 +1,17 @@
+from abc import abstractmethod
+from typing import Any
+
+from web_poet.pages import Injectable
+
+
+class Stats(Injectable):
+    """Stat collector."""
+
+    @abstractmethod
+    def set(self, key: str, value: Any) -> None:
+        """Sets the value of stat *key* to *value*."""
+
+    @abstractmethod
+    def inc(self, key: str, value: int = 1) -> None:
+        """Increments the value of stat *key* by *value*, or sets it to *value*
+        if *key* has no value."""
```

Comment:
Another idea (not sure if it's good). It seems we expect frameworks to return custom Stats objects, not instances of this class in particular. And that's different from all other dependencies, including HttpClient dependency. It seems this is what causes problems when trying to support it in tests.

Would it make sense to try another approach, where Stats dependency is not an abc, but there is a customizable stats storage, similar to how the concrete request downloader is customizable by HttpClient?

Reply:
Applied, along with the [“serialize to nothing, return dummy stats” approach discussed earlier](https://github.com/scrapinghub/web-poet/pull/175#issuecomment-1535973215).

Reply:
Nice! Looks much more straightforward.